### PR TITLE
Make goal aligner/command compatible with shooter

### PR DESCRIPTION
**Pull request by selym3**
* Change drivetrain goal alignment command to work with any `DrivetrainGoalAligner`
* Create `ShooterAligner` that extends from Goal aligner but uses a dynamic distance
* Change `DrivetrainAutoAlignment` to use `ShooterAligner`

### CODE REVIEW

File: src/main/java/com/stuypulse/robot/commands/DrivetrainAutomaticAlign.java [hunk 1]
```diff
@@ -5,24 +5,32 @@
...
+
+        // holds the shooter because the shooter mode / distance changes
+        private final Shooter shooter;
+
+        public ShooterAligner(Shooter shooter) {
+            super(0); // default distance of zero
+            this.shooter = shooter;
+        }
+
+        private Number getUpdatedDistance() {
```
Comment (Sam-Belliveau) [inline src/main/java/com/stuypulse/robot/commands/DrivetrainAutomaticAlign.java]: This function seems really redundant as its only called once.

I understand that it helps with readability, but its confusing so see this function defined here as it leads me to expect it to be used in multiple spots.

File: src/main/java/com/stuypulse/robot/commands/DrivetrainGoalCommand.java [hunk 2]
```diff
@@ -10,14 +10,18 @@
...
+    public DrivetrainGoalCommand(Drivetrain drivetrain, DrivetrainGoalAligner goalAligner) {
+        super(drivetrain, goalAligner);
+        setMaxSpeed(Constants.Alignment.Speed.LIMELIGHT_MAX_SPEED);
+        useMinTime();
+    }
+
     /**
      * @param drivetrain drivetrain to do movements on
      * @param distance distance to move from the target
      * @param innerGoal if we should aim for the inner goal
```
Comment (Sam-Belliveau) [inline src/main/java/com/stuypulse/robot/commands/DrivetrainGoalCommand.java]: these JavaDocs are old :/

File: src/main/java/com/stuypulse/robot/commands/DrivetrainGoalCommand.java [hunk 3]
```diff
@@ -10,14 +10,18 @@
 /** Command that aligns with the target */
 public class DrivetrainGoalCommand extends DrivetrainAlignmentCommand {
 
+    public DrivetrainGoalCommand(Drivetrain drivetrain, DrivetrainGoalAligner goalAligner) {
```
Comment (Sam-Belliveau) [inline src/main/java/com/stuypulse/robot/commands/DrivetrainGoalCommand.java:19]: add comments like the other functions

File: src/main/java/com/stuypulse/robot/commands/DrivetrainGoalCommand.java [hunk 4]
```diff
@@ -10,14 +10,18 @@
...
     /**
      * @param drivetrain drivetrain to do movements on
      * @param distance distance to move from the target
      * @param innerGoal if we should aim for the inner goal
      */
     public DrivetrainGoalCommand(Drivetrain drivetrain, Number distance) {
-        super(drivetrain, new DrivetrainGoalAligner(distance));
-        setMaxSpeed(Constants.Alignment.Speed.LIMELIGHT_MAX_SPEED);
-        useMinTime();
+        this(drivetrain, new DrivetrainGoalAligner(distance));
```
Comment (Sam-Belliveau) [inline src/main/java/com/stuypulse/robot/commands/DrivetrainGoalCommand.java:32]: `setMaxSpeed(Constants.Alignment.Speed.LIMELIGHT_MAX_SPEED);`
and
`useMinTime();`

are very important to this class aligning correctly.

**Review by Sam-Belliveau — CHANGES_REQUESTED** [review]
This one is important to merge but there are a couple issues.

File: src/main/java/com/stuypulse/robot/commands/DrivetrainGoalCommand.java (hunk 4 above)
Comment (selym3) [reply src/main/java/com/stuypulse/robot/commands/DrivetrainGoalCommand.java:32]: Those functions are called within the other constructor

**Pull request merged**